server.js

```diff
@@ -602,10 +615,29 @@ function SocketIOFileUploadServer(options) {
 			socket.on(self.topicName + "_progress", _uploadProgress(socket));
 			socket.on(self.topicName + "_done", _uploadDone(socket));
 		}
+		self.functions.disconnect[socket.id] = _onDisconnect(socket);
+		socket.on("disconnect", self.functions.disconnect[socket.id]);
+	};
 
-		socket.on("disconnect", _onDisconnect(socket));
+	/**
+	 * Public method.  Undo listening from listen method.
+	 *
+	 * @param  {Socket} socket The socket on which not to listen anymore
+	 * @return {void}
+	 */
+	this.close = function (socket) {
```

Comment:
I think you should call `_onDisconnect(socket)` in this function, because if there is an upload in progress, you need to clean up file handles.